### PR TITLE
CORE-5577 holding id cleanup

### PR DESCRIPTION
**Pull request by barrylapthorn**
Rename all the many, many, many variants to "holding identity short hash" (and PascalCase, and camelCase variants as well).

This fixes up "client experience" when using our REST apis, in other words - the _terminology is consistent in the customer facing REST API_.

Fix up swagger baseline and detekt errors.

After this PR it should be "somewhat obvious" where we can strongly type `holdingIdentityShortHash` in the code to further prevent random naming and passing incorrect parameters (particularly in unit tests).

NOTE:  this change will likely break systems outside of `corda-runtime-os` because the REST api has changed slightly (we _return_ the holding id short hash with a slightly different name).

```
// POST https://localhost:8888/api/v1/swagger#/VirtualNodeRPCOps/post_virtualnode)
{
...
  "holdingIdentityShortHash": "string",
...
}
```

```
// schema HoldingIdentity
fullHash* | string
groupId* | string
shortHash* | string
x500Name* | string
```

Example change as seen in the Swagger API:

image

image

image

### CODE REVIEW

File: libs/virtual-node/virtual-node-info/src/main/kotlin/net/corda/virtualnode/HoldingIdentity.kt [hunk 1]
```diff
@@ -10,14 +10,14 @@ data class HoldingIdentity(val x500Name: String, val groupId: String) {
      *
      * To be used as a short look-up code that may be passed back to a customer as part of a URL etc.
      */
-    val id: String by lazy(LazyThreadSafetyMode.PUBLICATION) {
-        hash.substring(0, 12)
+    val shortHash: String by lazy(LazyThreadSafetyMode.PUBLICATION) {
```
Comment (barrylapthorn) [inline libs/virtual-node/virtual-node-info/src/main/kotlin/net/corda/virtualnode/HoldingIdentity.kt:13]: Major rename

File: libs/virtual-node/virtual-node-info/src/main/kotlin/net/corda/virtualnode/HoldingIdentity.kt [hunk 2]
```diff
@@ -10,14 +10,14 @@ data class HoldingIdentity(val x500Name: String, val groupId: String) {
...
+        fullHash.substring(0, 12)
     }
 
     /**
      * Returns the holding identity full hash (SHA-256 hash of the x500 name and group ID)
      */
-    val hash: String by lazy(LazyThreadSafetyMode.PUBLICATION) {
+    val fullHash: String by lazy(LazyThreadSafetyMode.PUBLICATION) {
```
Comment (barrylapthorn) [inline libs/virtual-node/virtual-node-info/src/main/kotlin/net/corda/virtualnode/HoldingIdentity.kt:20]: Major rename

File: testing/virtual-node-info-read-service-fake/src/main/kotlin/net/corda/virtualnode/read/fake/VirtualNodeInfoReadServiceFake.kt [hunk 3]
```diff
@@ -112,9 +112,9 @@ class VirtualNodeInfoReadServiceFake internal constructor(
         return map[holdingIdentity]
     }
 
-    override fun getById(id: String): VirtualNodeInfo? {
+    override fun getByHoldingIdentityShortHash(holdingIdentityShortHash: String): VirtualNodeInfo? {
```
Comment (barrylapthorn) [inline testing/virtual-node-info-read-service-fake/src/main/kotlin/net/corda/virtualnode/read/fake/VirtualNodeInfoReadServiceFake.kt:115]: Significant rename (via the interface)

File: components/virtual-node/virtual-node-info-read-service/src/main/kotlin/net/corda/virtualnode/read/VirtualNodeInfoReadService.kt [hunk 4]
```diff
@@ -41,11 +41,11 @@ interface VirtualNodeInfoReadService : ReconcilerReader<HoldingIdentity, Virtual
      *
      * Implementation may return null if it is part of a component.
      *
-     * Short id can be generated by calling the [HoldingIdentity.id]
+     * Short id can be generated by calling the [HoldingIdentity.shortHash]
      *
      * Returns `null` if no such information exists
      */
-    fun getById(id: String): VirtualNodeInfo?
+    fun getByHoldingIdentityShortHash(holdingIdentityShortHash: String): VirtualNodeInfo?
```
Comment (barrylapthorn) [inline components/virtual-node/virtual-node-info-read-service/src/main/kotlin/net/corda/virtualnode/read/VirtualNodeInfoReadService.kt:48]: Significant rename

**Comment by driessamyn** [comment]
I think this change looks good and I propose we merge it despite a wide impact.

There is a follow-up task (https://r3-cev.atlassian.net/browse/CORE-5455) to turn shortHash into a custom type so the definition of it is fixed and consistent, and while this will mean another wide change, I think it makes sense to split it into 2.

@vkolomeyko, @JamesHR3 are you ok with this?

@davidcurrie FYI, this will impact the REST API, so will likely impact the soak tests

If everybody agrees, let's send out notice that we will make this change and then merge ASAP so we don't end up with too many rebases.

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by vkolomeyko — DISMISSED** [review]

**Review by JamesHR3 — DISMISSED** [review]
Makes sense to me, and I see we've started to get the comms out so I'm comfortable with sticking it in when we're happy that's had enough time to get around.

**Comment by barrylapthorn** [comment]
Pushed one more commit to tidy up the CPI file checksum parameter name as well to reduce "churn" that the QA teams might have if we were to put that commit in a second PR.

**Review by driessamyn — DISMISSED** [review]
LGTM.

Will re-run the build tomorrow afternoon to make sure it's still compatible with main then, and merge at the end of the day.

**Review by driessamyn — APPROVED** [review]
LGTM

**Pull request merged**